librosa/core/notation.py

```diff
@@ -122,16 +123,17 @@
 
 
 # Pre-compiled regular expressions for note and key parsing
+KEY_RE = re.compile(
+    r"^(?P<tonic>[A-Ga-g])" r"(?P<accidental>([#♯b!♭𝄪𝄫]*))" r":(?P<scale>(maj|min)(or)?)$"
+)
 NOTE_RE = re.compile(
     r"^(?P<note>[A-Ga-g])"
     r"(?P<accidental>[#♯𝄪b!♭𝄫♮]*)"
     r"(?P<octave>[+-]?\d+)?"
     r"(?P<cents>[+-]\d+)?$"
 )
 
-KEY_RE = re.compile(
-    r"^(?P<tonic>[A-Ga-g])" r"(?P<accidental>[#♯b!♭]?)" r":(?P<scale>(maj|min)(or)?)$"
-)
+ACC_MAP = {"#": 1, "": 0, "b": -1, "!": -1, "♯": 1, "♭": -1, "##": 2, "♯♯": 2, "𝄪": 2, "bb": -2, "♭♭": -2, "𝄫": -2}
```

Comment:
I wonder - if we're going to extend the accidental accumulator logic to support variable number of tokens, we can probably remove the explicit support for non-unicode doubles `##` and `bb`, right?

Also, natural should probably include here as mapping explicitly to 0 given the discussion elsewhere in this PR.